src/App.tsx

```diff
@@ -1,26 +1,228 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  FormEvent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { addTodo, changeTodo, deleteTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterOptions } from './enums/FilterOptions';
+import { ErrorMessages } from './enums/ErrorMessages';
+import { filteredTodos } from './utils/filteringTodos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { TodoList } from './components/TodoList';
+import { USER_ID } from './utils/USER_ID';
+import { Errors } from './components/Errors';
+import { loadTodos } from './utils/loadTodos';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterOptions>(FilterOptions.ALL);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessages.NO_ERRORS);
+  const [loader, setLoader] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodosCount, setLoadingTodosCount] = useState<number[]>([]);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [todos, inputRef]);
```

Comment:
Move the `useEffect `to the end of code
Placing `useEffect `at the end improves readability by letting developers first grasp the main logic before seeing side effects. This promotes clearer structure and consistency in the codebase.